src/vision/status_quo/alan_has_an_event_loop.md

```diff
@@ -0,0 +1,104 @@
+
+Once Alan is happy whit his Mock UI, he wants to hook it up with some actual features. He is happy to learn that Rust has Features he already knows.
+* Promises => Futures
+* Observables => Streams.
+
+Alan smiles, thinking he knows what and more importantly how to do this.
+
+### First time dealing with runtimes
+
+Unfortunately, coming from frameworks like Angular or Node.js, Alan is not used that he himself is responsible for driving the progress of Async/Streams. 
+
+After reading up about Runtimes, his mental image of a runtime is something like: 
+
+```rust
+impl Runtime {
+    fn run() {
+        while !self.tasks.is_empty() {
+            while let Some(task) = self.awoken_tasks.pop() {
+                task.poll();
+                //... remove finished task from 'tasks'
+            }
+        }
+    }
+}
+```
+
+Coming from Single-Threaded Angular development, Alan decides to limit his new App to Single-Threaded. He does not feel like learning about Send/Sync/Mutex as well as struggling with the borrow checker.
+
+On top of that, his App is not doing any heavy calculation so he feels async should be enough to not block the main thread too bad and have a hanging UI.
+
+### Fun time is over
+
+Soon Alan realises that he cannot use any of those runtimes because they all take control of the thread and block. The same as the OS Event loop.
+
+Alan spends quite some time to look through several runtime implementations. Ignoring most internal things, all he wants is a runtime that looks a bit like this:
+
+```rust
+impl Runtime {
+    fn make_progress() {
+        while let Some(task) = self.awoken_tasks.pop() {
+            task.poll();
+            //... remove finished task from 'tasks'
+        }
+    }
+    fn run() {
+        while !self.tasks.is_empty() {
+            self.make_progress();
+        }
+    }
+}
+```
+
+It could be soo easy. Unfortunately he does not find any such solution. Having already looked through quite a bit of low level documentation and runtime code, Alan thinks about implementing his own runtime...
+
+...but only for a very short time. Soon after looking into it, he finds out that he has to deal with ```RawWakerVTable```, ```RawWaker```, ```Pointers```. Worst of all, he has to do that without the safety net of the rust compiler, because this stuff is ```unsafe```.
+
+Reimplementing the OS Event Loop is also not an option he wants to take. See [here](https://developer.apple.com/documentation/appkit/nsapplication)
+>Override run() if you want the app to manage the main event loop differently than it does by default. (This a critical and complex task, however, that you should only attempt with good reason).
+
+
+### The cheap way out
+
+Alan gives up and uses a runtime in a seperate thread from the UI. This means he has to deal with the additional burden of syncing and he has to give up the frictionless use of some of the patterns he is accustomed to by treating UI events as ```Stream<Item = UIEvent>```.
```

Comment:
Can you say a bit more about what he has to do instead? What is the code he *wanted* to write, and what did he get stuck with? 

I'm guessing here the problem is that `UIEvent` can't be sent to other threads?